src/RawPOSIX/src/safeposix/dispatcher.rs

```diff
@@ -379,56 +535,104 @@ pub fn lind_syscall_api(
         }
 
         EXEC_SYSCALL => {
+            // NaCl equivalent: NaClSysExec
             let child_cageid = arg1 as u64;
+        
+            // Perform exec operation through cage implementation
+            // Child cage validation handled by cage layer
             interface::cagetable_getref(cageid)
                 .exec_syscall(child_cageid)
         }
 
         EXIT_SYSCALL => {
+            // NaCl equivalent: NaClSysExit
             let status = arg1 as i32;
+        
+            // Perform exit operation through cage implementation
+            // Cleanup handled by cage layer
             interface::cagetable_getref(cageid)
                 .exit_syscall(status)
         }
 
         SELECT_SYSCALL => {
+            // NaCl equivalent: NaClSysSelect
             let nfds = arg1 as i32;
-            let readfds = interface::get_fdset(arg2).unwrap();
-            let writefds = interface::get_fdset(arg3).unwrap();
-            let errorfds = interface::get_fdset(arg4).unwrap();
-            let rposix_timeout = interface::duration_fromtimeval(arg5).unwrap();
+            
+            // Get and validate fd sets
+            // NaCl: Uses NaClCopyInFromUser for fd_set validation
+            let readfds = match interface::get_fdset(arg2) {
```

Comment:
readfds (or arg2) is a pointer that points to fdset struct. So I think we also need to do things like `check_and_convert_addr_ext` for it. Same for writefds, errorfds and timeout

Reply:
Done